code/API_definitions/qod-api.yaml

```diff
@@ -279,7 +426,9 @@ components:
         applicationServerPorts:
           $ref: "#/components/schemas/PortsSpec"
         qosProfile:
-          $ref: "#/components/schemas/QosProfile"
+          oneOf:
```

Comment:
@jlurien
I assume the intention is to drop this support for the "legacy" QoS profiles at some stage, so `QosProfile` would be deleted and `qosProfile` would then be of type `QoSProfileId` only. If that is the intention, then we should keep the restriction that `QoSProfileId` be a uuid (I agree with your comment below) and live with this workaround until it is agreed that the current QoS profiles are no longer useful.

If that is not the intention, and the legacy profiles must always be supported, then simply redefining `QoSProfileId` to have `minLength: 5` would achieve that, albeit at the cost of more complex validation.

Reply:
To me the [proposal](https://github.com/camaraproject/QualityOnDemand/issues/125#issuecomment-1520183394) by @sfnuser makes sense. I don't see why the current predefined qosProfiles should be treated differently once we have the new discovery functionality in place.

We should also decide if API spec enforce some predefined QoS Profile to be offered by all implementations, and what is the best unique identifier for a profile, a UUID or some more friendly label such as current QOS_X. Problem with UUID is that they are not friendly for predefined profiles.

Reply:
> To me the [proposal](https://github.com/camaraproject/QualityOnDemand/issues/125#issuecomment-1520183394) by @sfnuser makes sense. I don't see why the current predefined qosProfiles should be treated differently once we have the new discovery functionality in place.

I can see going down this path of not using a UUID for the QoS profiles and instead use a name as the id.  We can define this name to be a unique string that matches this regex `^[a-zA-Z0-9_]+$`

The length of the name can be from 1 to 128 characters?

With this, the name would replace the ID as the unique identifier.

I'd also like to add an optional `displayName` attribute as a string that could be easier to read, and can contain spaces and other special characters. This would replace the current name attribute and would not be unique.

Since the QOS_S etc. can be a part of this, I think that all of the other attributes would become optional without default values.  This is due to the fact that QOS_S etc. can be defined by only using the name field.

Reply:
One more note. Using a name with some constraints is consistent with the way amazon implemented their EC2 instance types. EC2 instance sizing is a rough analogy for the QoSProfiles. https://aws.amazon.com/ec2/instance-types. So this pattern should be familiar to many application developers.